Enhancement: added X-UAV Mini Talon V-Tail airframe

fredowski opened the pull request:
The X-UAV Mini Talon airframe with the
new type "Plane V-Tail" uses the AAVVTWFF mixer
and is derived from the Albatross A-Tail.

This airframe introduces a new airframe group "Plane V-Tail". There
is a corresponding PR mavlink/qgroundcontrol#6852 to introduce the
new airframe group in qgroundcontrol with a picture.

We use the X-UAV Mini Talon in our [research project](https://www.hs-augsburg.de/homes/beckmanf/dokuwiki/doku.php?id=searchwing) and I plan to maintain
the parameters. We have done several flights with these parameters for example
[this one](https://review.px4.io/plot_app?log=0af78132-95a9-4260-aa7b-19f211b62aa2) although they are simply copied from the Albatros A-Tail. Nothing is optimized, but it flies.

fredowski commented:
I checked the setup again at our X-UAV Mini Talon and I found that the two V-Tail outputs are flipped in the hardware. I prepare and test an update because I want to keep the outputs as they are described. I think I will need a dedicated mixer because this will not work with simply inverting the servos.

dagar commented:
Looks safe to merge, but what do you think about keeping the mixer generic? It helps on older platforms where flash space is getting tight.

fredowski commented:
Hi Daniel, thanks for looking at the patch.

Actually the mixer is generic. We used the AAVVTW mixer from the Albatros A-Tail but in order to use it we had to reverse the servos and flip the left and right V-Tail outputs. I noticed the V-Tail output flip only after checking again against the documentation. In addition the flaperon worked in the wrong direction. We do not use the flaperon so I figured that out only when I finally tested the mixer against documentation. So it turned out that we cannot keep the AAVVTWFF order at the outputs and keep the AAVVTWFF Albatros mixer plus servo inversion. That is the reason why I introduced the new mixer and removed the servo inversions. Compared to the Albatros mixer I only changed signs not the mixing values. I also left the wheel and flaps mixing outputs although the mini talon does not have a steerable wheel or flaps.

The reason for keeping the output order identical is my personal experience when we started with the Bixler3 standard tail and we tried out existing plane models. It took a while before I noticed that some plane models have a different output order (AERT vs. AETRFG mixer). At least for us that was quite confusing - for us the AERT word did not have any meaning...

fredowski commented:
Hi David,

we did a test flight with the proposed X-UAV Mini talon frame including the mixer. So that is exactly this patch. See:

https://review.px4.io/plot_app?log=d08a98d1-d7da-434c-90e5-0a0cf7a30860

Cheers!

Friedrich

fredowski commented:
I updated the patch to reflect the generic v-tail mixer type.

dagar commented:
Minor error in the airframe script. See the CI failure.
http://ci.px4.io:8080/job/PX4/job/Firmware/job/PR-10414/7/display/redirect

![image](https://user-images.githubusercontent.com/84712/46806835-9d8d2080-cd36-11e8-833d-19bffa6d68ab.png)

This is done to make the shell script compatible with sh, bash, nsh, etc.

dagar commented:
@Tony3dr this will show up in QGC soon.

Tony3dr commented:
@dagar, Great Thanks!

Avysuarez commented:
Hi!
I just configured with this new frame and it does not recognize the radio. I had been using the albatross parameters and i had good flights with that configuration.

fredowski commented:
It does not recognize the radio? How can that be related to the airframe?

Avysuarez commented:
not with the fuselage, but witht the parameters of that airframe, because I loaded the albatross airframe again and it alreday recognizes the radio.

fredowski commented:
Mmh. Could you maybe
- Close QGC
- Delete ~/.config/org.qgroundcontrol
- Restart QGC
and try again? I sometimes had problems when I switched between airframes.

Avysuarez commented:
ok, I'll do it
